### PR TITLE
External RMI Tunneling with Traefik

### PR DESCRIPTION
**Pull request by anpanigr**
Added External RMI Tunneling with Traefik/Nginx  in addition to existing Voyager.

Modification to  ItExternalRmiTunneling.java

a. Move the common certificate task to beforeAll()
b. add new method to test JMS Client outside K8s  so that it can be used for different LB usecase.

Jenkin 
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5159/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5201

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java [hunk 1]
```diff
@@ -465,6 +599,68 @@ public void testExternalRmiAccessThruHttpsTunneling() {
         .until(runJmsClient(new String(javasCmd)));
   }
 
+  // Run the RMI client inside K8s Cluster
+  private void runPodClient(int serverCount, boolean checkConnection) {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java:603]: change method name to runClientInsidePod

**Review by vanajamukkara — COMMENTED** [review]
move nginx code to test method until the test is enabled, otherwise looks good.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java [hunk 2]
```diff
@@ -269,112 +306,119 @@ public void testExternalRmiAccessThruHttpTunneling() {
...
-    javapCmd.append(" JmsTestClient ");
-    javapCmd.append(" t3://");
-    javapCmd.append(domainUid);
-    javapCmd.append("-cluster-");
-    javapCmd.append(clusterName);
-    javapCmd.append(":8001 2 true");
-    javapCmd.append(" \"");
-    logger.info("java command to be run {0}", javapCmd.toString());
+    // works inside pod
+    // runPodClient(2, true);
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java]: Removed commented code

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java (hunk 1 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java]: modified the method name

**Review by vanajamukkara — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java [hunk 3]
```diff
@@ -435,6 +464,108 @@ public void testExternalRmiAccessThruHttpsTunneling() {
     assertTrue(httpsTunnelingPort != -1,
         "Could not get the HttpsTunnelingPort service node port");
     logger.info("HttpsTunnelingPort for Voyager {0}", httpsTunnelingPort);
+    runExtHttpsClient(httpsTunnelingPort, 2, false);
+  }
+
+  @Order(5)
+  @Test
+  @DisplayName("Verify tls RMI access WLS through Traefik loadBalancer")
+  public void testExternalRmiAccessThruTraefikHttpsTunneling() {
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java:473]: missing javadoc

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java [hunk 4]
```diff
@@ -435,6 +464,108 @@ public void testExternalRmiAccessThruHttpsTunneling() {
...
+        getServiceNodePort(traefikNamespace, service, "websecure");
+    assertTrue(httpsTunnelingPort != -1,
+        "Could not get the Traefik HttpsTunnelingPort service node port");
+    logger.info("HttpsTunnelingPort for Traefik {0}", httpsTunnelingPort);
+    runExtHttpsClient(httpsTunnelingPort, 2, false);
+  }
+
+  @Disabled("NGNIX tls ingress yaml file not ready")
+  @Order(6)
+  @Test
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java:519]: missing javadoc

Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java:531]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java (hunk 3 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java:482]: done

**Pull request merged**